### PR TITLE
Fix missing mask on tl.atomic_add for constant-index stores

### LINKED ISSUE #178871: Incorrect output for non-power-of-2-sized fusion of scalar broadcast and scatter-add
### 🐛 Describe the bug

When inductor fuses a scalar broadcast with `scatter_reduce(..., reduce="sum")`, the generated Triton kernel uses `mask=None` on `tl.atomic_add` instead of `mask=xmask`. `XBLOCK` is always a power of 2, so when `N` is not a power of 2 there are `XBLOCK - N` out-of-bounds threads — they each add the broadcast scalar to the output, giving a wrong result with no error raised.

The error is exactly `(next_power_of_2(N) - N) * scalar` and is zero whenever `N` is a power of 2.

**Minimal reproducer:**

```python
import torch


def f(x, bias, idx):
    src = x + bias  # scalar broadcast fused with scatter by inductor
    out = torch.zeros(1, 1, device=x.device, dtype=x.dtype)
    return out.scatter_reduce(
        0, idx.unsqueeze(-1).expand_as(src), src, "sum", include_self=True
    )


for N in range(1, 65):
    device = torch.device("cuda")

    x    = torch.randn(N, 1, device=device)
    bias = torch.tensor([0.020368], device=device)
    idx  = torch.zeros(N, dtype=torch.long, device=device)

    ref = f(x, bias, idx)
    torch.compiler.reset()
    test = torch.compile(f, backend="inductor", fullgraph=True, dynamic=False)(x, bias, idx)

    diff      = test.item() - ref.item()
    next_pow2 = 1 << (N - 1).bit_length()
    expected  = (next_pow2 - N) * bias.item()
    print(f"N={N:2d}  diff={diff:+.6f}  expected={expected:+.6f}")
```

The generated kernel (trimmed):

```python
tmp0 = tl.load(in_ptr0 + x0, xmask)  # x     — masked, 0.0 for OOB threads
tmp1 = tl.load(in_ptr1 + 0)           # bias  — scalar, no mask needed
tmp2 = tmp0 + tmp1                    # bias for OOB threads (tmp0 == 0.0)

tl.atomic_add(out_ptr0 + ..., tmp2, None,  sem='relaxed')  # BUG: should be xmask
```

OOB threads add `bias` unconditionally. Total error: `(XBLOCK - N) * bias`.

### Error logs

No exception — the output is silently wrong.

```
N= 1  diff=+0.000000  expected=+0.000000
N= 2  diff=+0.000000  expected=+0.000000
N= 3  diff=+0.020368  expected=+0.020368
N= 4  diff=+0.000000  expected=+0.000000
N= 5  diff=+0.061104  expected=+0.061104
N= 6  diff=+0.040736  expected=+0.040736
N= 7  diff=+0.020368  expected=+0.020368
N= 8  diff=+0.000000  expected=+0.000000
N= 9  diff=+0.142576  expected=+0.142576
...
N=16  diff=+0.000000  expected=+0.000000
N=17  diff=+0.305520  expected=+0.305520
...
N=32  diff=+0.000000  expected=+0.000000
N=33  diff=+0.631407  expected=+0.631408
...
N=48  diff=+0.325887  expected=+0.325888
...
N=64  diff=+0.000000  expected=+0.000000
```

### Versions

```
PyTorch version: 2.10.0+cu128
Is debug build: False
CUDA used to build PyTorch: 12.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 12.3.0-1ubuntu1~22.04.2) 12.3.0
Python version: 3.11.15 | packaged by conda-forge | (main, Mar  5 2026, 16:45:40) [GCC 14.3.0] (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 12.8.93
GPU models and configuration: GPU 0: NVIDIA TITAN V
Nvidia driver version: 570.195.03
triton: 3.6.0
```

cc @ezyang @gchanan @kadeng @msaroufim @pragupta @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @muchulee8 @amjames @aakhundov @coconutruben @jataylo

### PR DESCRIPTION
**Pull request by shino16**
Fixes #178871

When a `scatter_reduce`'s output has size 1 in every dimension, `_fixed_indexer` skips all size-1 dims and the store index collapses to a constant integer. The [constant-index path](https://github.com/pytorch/pytorch/blob/cd2590172dbda49308d77a5cc17a30b87c97e42b/torch/_inductor/codegen/triton.py#L3352-L3378) in `TritonKernel.indexing()` skips mask computation for non-fixed-config kernels, so OOB threads accumulate stale values. See `mask_vars = OrderedSet()` below:

https://github.com/pytorch/pytorch/blob/cd2590172dbda49308d77a5cc17a30b87c97e42b/torch/_inductor/codegen/triton.py#L3371-L3378

This PR adds a `force_mask` parameter to `indexing()` that prevents the constant-index optimization from omitting the mask. `store()` passes `force_mask=True` for `atomic_add`.

## Test plan

New `test_scatter_reduce_fused_broadcast_non_power_of_2` in `CommonTemplate`, which is essentially a copy of the repro, verifies scatter_reduce + scalar broadcast fusion for several non-power-of-2 input sizes (3, 5, 7, 9, 17, 33, 48).

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @mlazos

### CODE REVIEW

File: torch/_inductor/codegen/triton.py [hunk 1]
```diff
@@ -3951,6 +3952,7 @@ def store(
             dense_indexing=True,
             block_ptr=mode is None,
             tma_compatibility_checker=tma_compatibility_checker,
+            mask_constant_index=mode == "atomic_add",
```
Comment (jansel) [inline torch/_inductor/codegen/triton.py:3955]: Will this be an issue for the other atomic ops too?

**Review by jansel — APPROVED** [review]

File: torch/_inductor/codegen/triton.py (hunk 1 above)
Comment (shino16) [reply torch/_inductor/codegen/triton.py:3955]: Yes if other atomic ops were implemented to `TritonKernel.store()`. As far as I checked, Inductor's backends only care `mode` in None, `atomic_add` or `tma`.
